src/uu/join/src/join.rs

```diff
@@ -60,13 +61,104 @@ enum FileNum {
     File2,
 }
 
-#[derive(Copy, Clone, PartialEq)]
-enum Sep {
-    Char(u8),
+#[derive(Clone)]
+enum SepSetting {
+    Byte(u8),
+    Char(Vec<u8>),
     Line,
     Whitespaces,
 }
 
+trait Separator: Clone {
```

Comment:
please add some comments

Reply:
Added doc comments to the associated methods and the instantiations. No comment for the trait itself since it is what it says (any join separator), but I can add one if you'd like.